tests/x11/vnc_two_passwords.pm

```diff
@@ -104,6 +105,12 @@ sub run {
 
         # Close xev
         send_key 'ctrl-c';
+        # repeat send_key because we have sporadic performance issue on ppc64le
+        # and wait a little more time for closing xev
+        if (is_ppc64le) {
+            send_key 'ctrl-c';
+            wait_still_screen 20;
+        }
```

Comment:
That looks wong to me.

Couldn't it be?:
```perl
# Close xev
do {
  # repeat send_key because we have sporadic performance issue on ppc64le
  # and wait a little more time for closing xev
  send_key 'ctrl-c';
  wait_still_screen;
} while (!check_screen('generic-desktop', 0));
```
That will be applied to all cases, but it will only be repeated if the windows is not closed.
Also, a `wait_still_screen` has the same danger as a sleep. Are you sure that 20 seconds is necessary?

Reply:
So, let's not reinvent the wheel, that should be a `send_key_until_needlematch`

Reply:
this problem here:

1. even if xev windows is gone, but it's still alive in background for a while, that means xev.log is not empty and it fails, this is sporadic happening atm.
2. 20 sec is not much for worker with poor performance

So send_key_until_needlematch doesn't work in this case, I already tested.
So I think we should give more time for it.

Reply:
https://openqa.suse.de/tests/4097990#step/vnc_two_passwords/23 shows the problem clearly by using:

 # Close xev
        do {
            # repeat send_key because we have sporadic performance issue on ppc64le
            # and wait a little more time for closing xev
            send_key 'ctrl-c';
            wait_still_screen;
        } while (check_screen('vncviewer-xev', 0));

Reply:
I know see why that logic doesn't work, but still your proposal looks wrong. `ctrl-c` should not be sent until the window is closed. It needs a sync point before sending the key, not repeating it again and wait.